### PR TITLE
Feature/sof 6234 - add unit clone without flowchartId

### PR DESCRIPTION
**Pull request by timurbazhirov**

### CODE REVIEW

File: src/units/base.js [hunk 1]
```diff
@@ -71,7 +73,14 @@ export class BaseUnit extends NamedDefaultableRepetitionRuntimeItemsImportantSet
      * @returns Boolean
      */
     isInStatus(status) {
-        return (this.status === status);
+        return this.status === status;
     }
 
+    clone(extraContext) {
+        const flowchartIDOverrideConfigAsExtraContext = {
+            flowchartId: BaseUnit.defaultFlowchartId(),
```
Comment (hcharley) [inline src/units/base.js:81]: Can possibly be changed to `generateFlowChartId()`

File: tests/unit.test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { createWorkflows } from "../src/workflows";
+
+describe("units", () => {
+    it("can be cloned with new flowchartId", () => {
+        const workflows = createWorkflows({});
+        const exampleWorkflow = workflows[0];
+        const exampleSubworkflow = exampleWorkflow.subworkflows[0];
+        const exampleUnit = exampleSubworkflow.units[0];
+        const exampleUnitClone = exampleUnit.clone();
+        expect(exampleUnitClone).to.exist;
```
Comment (hcharley) [inline tests/unit.test.js:12]: Does `exist` need to be called as `exist()`?

**Review by hcharley — APPROVED** [review]

File: tests/unit.test.js (hunk 2 above)
Comment (timurbazhirov) [reply tests/unit.test.js:12]: Looks like not

File: src/units/base.js (hunk 1 above)
Comment (timurbazhirov) [reply src/units/base.js:81]: TBD in a followup task

Comment (timurbazhirov) [reply src/units/base.js:81]: https://exabyte.atlassian.net/browse/SOF-6182

**Pull request merged**